src/Sort.c

```diff
@@ -144,6 +156,182 @@ int quickSort(int arr[], int low, int high){
+  // first choose middle of array
+  int middle;
+  // if size is even, middle is offset by 1 to the right
+  if (size % 2 == 0) {
+    middle = size / 2;
+  } else {
+    middle = (size - 1) / 2;
+  }
+  if (size > 0) {
+    // check if selected middle is already the target element
+    if ((arr+middle)->number == target) {
+      printf("Found it!\n");
+      printf("Number: %d - Letters: %s\n", (arr+middle)->number, (arr+middle)->letters);
+    } // if target is bigger than middle call searchForNumber on right half of the array
+    else if ((arr+middle)->number < target) {
+      searchForNumber(arr+middle, target, size - middle);
+    } // if target is smaller than middle call searchForNumber on left half of the array
+    else if ((arr+middle)->number > target) {
+      searchForNumber(arr, target, middle);
+    }
+  }
+  return 0;
+}
+
+// searches for given string sequence in array of Element of given size already sorted by letters
+// starts searching in the middle of array and goes left or right depending on target element
+int searchForLetters(Element arr[], char target[], int size) {
+  // first choose middle of array
+  int middle;
+  // if size is even, middle is offset by 1 to the right
+  if (size % 2 == 0) {
+    middle = size / 2;
+  } else {
+    middle = (size - 1) / 2;
+  }
+  if (size > 0) {
+    // check if selected middle is already the target element
+    if (strcmp((arr+middle)->letters, target) == 0) {
+      printf("Found one!\n");
+      printf("Number: %d - Letters: %s\n", (arr+middle)->number, (arr+middle)->letters);
+    } else if (isBiggerOrSameCharset(target, (arr+middle)->letters) == 0) {
+      searchForLetters(arr+middle, target, size - middle);
+      //printf("Went to the right\n");
+    } else if (isBiggerOrSameCharset((arr+middle)->letters, target) == 0) {
+      //printf("Went to the left\n");
+      searchForLetters(arr, target, middle);
+    }
+  }
+  return 0;
+}
+
+// continiously searches for input from stdin for either a given number or given string sequence in array of Element
+int searchForInput(Element arr[], int low, int high) {
+  int input = 1;
+  while (input != 0) {
+    printf("Do you want to search for numbers (type '1') or letters (type '2'):\nOr press '0' to quit.\n");
+    scanf("%d", &input);
+
+    switch (input) {
+    case 1:;
+      int number = 0;
+      printf("Type in the number you want to search for:\n");
+      scanf("%d", &number);
+      quickSortWithElements(arr, low, high-1, "n");
+      searchForNumber(arr, number, high);
+      break;
+    case 2:
+      printf("Type in the letters (3 upper-case letters only) you want to search for:\n");
+      char letters[3];
+      scanf("%s", letters);
+      quickSortWithElements(arr, low, high-1, "l");
+      searchForLetters(arr, letters, high);
+      break;
+    case 0:
+      printf("Quitting...\n");
+      break;
+    default:
+      fprintf(stderr, "Error: Did not provide valid input - choose either '1' or '2'\n");
```

Comment:
...or 0 to quit"
return 1 here prevents that the user is prompted again for input?

Reply:
you are totally right - but I don't get what's happening...